spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/AppRegistryCommands.java

```diff
@@ -248,7 +249,13 @@ public String importFromResource(
 			try {
 				Resource resource = this.resourceLoader.getResource(uri);
 				Properties applications = PropertiesLoaderUtils.loadProperties(resource);
-				PagedResources<AppRegistrationResource> registered = appRegistryOperations().registerAll(applications, force);
+				PagedResources<AppRegistrationResource> registered = null;
+				try {
+					registered = appRegistryOperations().registerAll(applications, force);
+				}
+				catch (Exception e) {
+					return "Error when registering applications from " + uri + ": " + e.getMessage();
```

Comment:
I think the REST controller advice for the exceptions that are caught at the REST controller side would still be better in this case. In any case, we can add it later if we find it necessary.